File: common/data/src/main/java/org/thingsboard/server/common/data/id/RpcId.java
package org.thingsboard.server.common.data.id;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.annotations.ApiModelProperty;
import org.thingsboard.server.common.data.EntityType;

import java.util.UUID;

public final class RpcId extends UUIDBased implements EntityId {

    private static final long serialVersionUID = 1L;

    @JsonCreator
    public RpcId(@JsonProperty("id") UUID id) {
        super(id);
    }

    @ApiModelProperty(position = 2, required = true, value = "string", example = "RPC", allowableValues = "RPC")
    @Override
    public EntityType getEntityType() {
        return EntityType.RPC;
    }
}

File: ui-ngx/src/app/shared/models/id/rpc-id.ts
import { EntityId } from '@shared/models/id/entity-id';
import { EntityType } from '@shared/models/entity-type.models';

export class RpcId implements EntityId {
  entityType = EntityType.RPC;
  id: string;
  constructor(id: string) {
    this.id = id;
  }
}
